### PR TITLE
make grpc-encoding:identity implicit

### PR DESCRIPTION
**Pull request by dgquintas**
@y-zeng for the modified test (the change surfaced a deficiency in it).

### CODE REVIEW

File: test/core/end2end/bad_server_response_test.c [hunk 1]
```diff
@@ -82,7 +82,7 @@
 #define HTTP1_DETAIL_MSG "Trying to connect an http1.x server"
 
 /* TODO(zyc) Check the content of incomming data instead of using this length */
-#define EXPECTED_INCOMING_DATA_LENGTH (size_t)310
+#define EXPECTED_INCOMING_DATA_LENGTH (size_t)300
```
Comment (ctiller) [inline test/core/end2end/bad_server_response_test.c]: 200?

If it's a lower bound, let's make it a lower bound we won't quickly hit.

Comment (dgquintas) [inline test/core/end2end/bad_server_response_test.c]: @y-zeng can you shed some light on the role this plays in the test? The naming is a bit misleading currently.

Comment (y-zeng) [inline test/core/end2end/bad_server_response_test.c]: The 'bad' server will start sending responses after reading this amount of data from the client. Sorry for the confusion :anguished:

Comment (dgquintas) [inline test/core/end2end/bad_server_response_test.c]: I've renamed the define and lowered the threshold. PTAL.

**Comment by dgquintas** [comment]
jenkins test this please

**Comment by dgquintas** [comment]
Issues #9752 #9620 #9738

**Pull request merged**